### PR TITLE
Spells: Allow secondary target checking for persistent area auras

### PR DESCRIPTION
**Pull request by insunaa**
## 🍰 Pullrequest

This PR allows for checking for secondary targets in an AuraScript

### Todo / Checklist

- [ ] Sanity check
- [ ] Missing pieces?

### CODE REVIEW

File: src/game/Grids/GridNotifiersImpl.h [hunk 1]
```diff
@@ -284,6 +284,8 @@ inline void MaNGOS::DynamicObjectUpdater::VisitHelper(Unit* target)
         if (!holder->GetAuraByEffectIndex(eff_index))
         {
             PersistentAreaAura* Aur = new PersistentAreaAura(spellInfo, eff_index, &i_dynobject.GetDamage(), &i_dynobject.GetBasePoints(), holder, target, caster);
+            if (!i_dynobject.OnAreaAuraCheckTarget(Aur, target))
```
Comment (killerwife) [inline src/game/Grids/GridNotifiersImpl.h:287]: Pass dyngo as param and do the check way higher before any aura creation. Also, you just wrote a memleak :D

Comment (insunaa) [reply src/game/Grids/GridNotifiersImpl.h:287]: Yeah, forgot that pointers don't go out of scope :D been rusty

File: src/game/Grids/GridNotifiersImpl.h [hunk 2]
```diff
@@ -258,6 +258,8 @@ inline void MaNGOS::DynamicObjectUpdater::VisitHelper(Unit* target)
         }
     }
 
+    i_dynobject.OnAreaAuraCheckTarget(target);
+
```
Comment (killerwife) [inline src/game/Grids/GridNotifiersImpl.h:262]: perhaps put a return after?

Comment (insunaa) [reply src/game/Grids/GridNotifiersImpl.h:262]: oops yeah brain afk

Comment (insunaa) [reply src/game/Grids/GridNotifiersImpl.h:262]: sorted

Comment (insunaa) [reply src/game/Grids/GridNotifiersImpl.h:262]: no wait, wrong branch sec

File: src/game/Spells/Scripts/SpellScript.h [hunk 3]
```diff
@@ -125,6 +125,8 @@ struct AuraScript
     virtual void OnPeriodicTickEnd(Aura* /*aura*/) const {}
     // called on persistent area aura dyngo lifetime end
     virtual void OnPersistentAreaAuraEnd(DynamicObject* /*dynGo*/) const {}
+    // called on AreaAura target selection
+    virtual bool OnAreaAuraCheckTarget(DynamicObject* /*dynGo*/, Unit* /*target*/) const { return true; }
```
Comment (killerwife) [inline src/game/Spells/Scripts/SpellScript.h:129]: OnPersistentAreaAuraCheckTarget

Comment (insunaa) [reply src/game/Spells/Scripts/SpellScript.h]: done

**Pull request merged**